src/components/Summary.tsx

```diff
@@ -1,9 +1,164 @@
-const Summary: React.FC<{ genreData: Map<string, number> }> = ({ genreData }) => {
-    console.log(genreData);
-    return (
-        <>
-        </>
-    );
-}
-
-export default Summary;
\ No newline at end of file
+import { Fragment, useEffect, useRef, useState } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+import { ResponseData } from "../interfaces/songInterface";
+import { ArtistData } from "../interfaces/artistInterface";
+import { CardData } from "../interfaces/cardInterface";
+import { getArtistInfo, getToken, tallyArtist } from "@/functions/apiFunctions";
+import Image from "next/image";
+
+const Summary: React.FC<ResponseData & CardData> = ({ items, genreData, dataRange }) => {
+  const [open, setOpen] = useState(false);
+  const backButtonRef = useRef(null);
+  const topThreeSongs = items.slice(0, 3);
+
+  const [artistData, setArtistData] = useState<ArtistData>({
+    genres: [],
+    href: "",
+    id: "",
+    images: [],
+    name: "",
+    popularity: 0,
+    type: "",
+    uri: "",
+  });
+
+  let topArtistHref = "";
+  const artistCounts = tallyArtist({ items });
+  const unsortedArtists = Array.from(artistCounts.entries());
+  const sortedArtists = new Map(Array.from(unsortedArtists).sort((a, b) => b[1] - a[1]));
+
+  const unsortedArray = Array.from(genreData.entries());
+  const sortedByValue = new Map(
+    Array.from(unsortedArray)
+      .sort((a, b) => b[1] - a[1])
+      .splice(0, 3)
+  );
+
+  items.map((item) => {
+    item.artists.map((artist) => {
+      if (artist.name === sortedArtists.entries().next().value[0]) {
+        topArtistHref = artist.href;
+      }
+    });
+  });
```

Comment:
All this will be run every time the component rerenders, and seems like if the array of items has hundreds of values can be very computationally expensive. Might be good to limit this to only recalculate when `items` is updated.

Reply:
moved it into another function inside the useEffect hook and set it so it only triggers a re-render when items is updated